students/TinaB/lesson04/mailroom_part2.py

```diff
@@ -0,0 +1,231 @@
+                              30, 60, 65982.55], "donate_times": [3]},
+                          {"firstname": "Jeff", "lastname": "Bezos",
+                              "donations": [52636.27], "donate_times": [1]},
+                          {"firstname": "Paul", "lastname": "Allen",
+                              "donations": [877.33, 22], "donate_times": [2]},
+                          {"firstname": "Steven", "lastname": "Hawking", "donations": [
+                              326892.24, 123, 123.33, 123, 123], "donate_times": [5]},
+                          {"firstname": "Justin", "lastname": "Timberlake", "donations": [999658.25, 1233, 123], "donate_times": [3]}]
+
+# Menus
+def menu_selection(prompt, dispatch_dict):
+    while True:  # loop forever until they quit
+        response = input(prompt)
+        if dispatch_dict[response]() == "exit menu":
+            break
+
+
+def single_print_sub_menu():
+    menu_selection(single_print_sub_prompt, single_print_sub_dispatch)
+
+
+def quit():
+    print("Quitting this menu now")
+    return "exit menu"
+
+# functions for prompts ----
+
+
+def get_name_input():
+    # Function to select user input to return to print function
+    user_input = input("\nLookup donor by: \n"
+                       "\t1. Firstname: \n"
+                       "\t2. Lastname:\n"
+                       "Please enter 1 or 2 >>  ")
+
+    donor_return = []
+    donor_check = 0
+    if int(user_input) == 1:
+        fname = input(
+            "Type the first name of the donor you are looking for:  ")
+    else:
+        lname = input("Type the last name of the donor you are looking for:  ")
+
+    for donor in donors_list_dictionary:
+        # print(donor["firstname"].lower())
+        if int(user_input) == 1:
+            if fname.lower() in donor["firstname"].lower():
+                donor_check = input("Is this the donor you are looking for: {} {}?  Please enter yes or no:  ".format(
+                    donor["firstname"], donor["lastname"]))
+                if donor_check == "yes":
+                    donor_return.append(donor["firstname"])
+                    donor_return.append(donor["lastname"])
+                    donor_check = 1
+                    break
+            else: # firstname is not in list
+                print("HERE!")
+                first_name = fname
+                last_name = input(
+                    "No donor with that first name found in donor list. Please provide a last name to add donor to list: ")
+                donor_return.append(first_name)
+                donor_return.append(last_name)
+                break
+
+        else:  # user input = 2
+            if lname.lower() in donor["lastname"].lower():
+                donor_check = input(
+                    "Is this the donor you are looking for:  {} {}?   Please enter yes or no:  ".format(donor["firstname"], donor["lastname"]))
```

Comment:
Watch line length for Pep8 here. It's acceptable to add line break at opening parens like this:
```
donor_check = input(
                    "Is this the donor you are looking for:  {} {}?   Please enter yes or no:  ".format(
                        donor["firstname"], donor["lastname"]))
```